### DISCUSSION TITLE
Would this compile for ARM architectures and for RISC-V?

### DISCUSSION THREAD

**Discussion by Raj2032** [robertknight/ocrs#75] [Q&A]
Ultimately I plan to use openCV for live video feed and process all the text through ocrs running on a raspberry pi or a similar mini computer open board using a risc-v architecture such as [STAR64](https://pine64.com/product/star64-model-a-8gb-single-board-computer/).

Not too sure if those processor architectures support ocrs as it uses certain features such as AVX or other stuff?

**Comment by robertknight** [comment]
Ocrs works on Arm. On that platform it uses [Arm Neon](https://www.arm.com/technologies/neon) instead of AVX. I have run it on a Raspberry Pi, just make sure you use a 64bit OS. It should work on RISC-V in theory, but it won't take advantage of the CPU's SIMD features, so it would run more slowly than an "equivalent" Arm system.

**Reply by Raj2032** [reply]
> I have run it on a Raspberry Pi

Interesting, how long did it take for it to process the image?

> should work on RISC-V in theory, but it won't take advantage of the CPU's SIMD features, so it would run more slowly than an "equivalent" Arm system.

Is there something planned to take advantage of the CPU's SIMD features in the near future?

**Comment by robertknight** [comment]
> Interesting, how long did it take for it to process the image?

On a Raspberry Pi 5 it takes about 4 seconds for [this test image](https://upload.wikimedia.org/wikipedia/commons/2/27/Poster_-_Gone_With_the_Wind_01.jpg). It will be quicker for an image with less text. I think there is quite a lot of room to optimize that in future.

> Is there something planned to take advantage of the CPU's SIMD features in the near future?

Not in the near future. RISC-V SIMD works a bit differently than AVX / Arm Neon because the size of vectors (ie. how many values can be processed in one operation) is not known at compile time, and the developers working on the Rust language and standard library are still figuring out how to deal with it.

**Reply by Raj2032** [reply]
I may just use `match_template` instead of ocr then cause I would like a much faster react time (not blaming you) as I know ocr operations are slower than matching templates using opencv. Anyways thanks very much for your entire support, I wish you the best of luck with this project :)
